### PR TITLE
GitHub api setup

### LINKED ISSUE #1: GitHub API: PRs
As a visitor or an admin user
When I visit any page on the site
I see the number of merged PRs across all team members
This number is updated as each member of the team merges more PRs

### LINKED ISSUE #2: GitHub API: Commits
As a visitor or an admin user
When I visit any page on the site
I see the number of commits next to each Github username
This number is updated as each member of the team contributes more commits

### LINKED ISSUE #3: GitHub API: User Names
As a visitor or an admin user
When I visit any page on the site
I see the Github usernames of myself and my teammates somewhere on the site
The most up to date usernames are always displayed

### LINKED ISSUE #4: GitHub API: Repo Name
As a visitor or an admin user
When I visit any page on the site
I see the name of the Github repo somewhere on the site

### PR DESCRIPTION
**Pull request by LawrenceWhalen**
## Changes
- Add API reader to services to pull from github
- Create Poros to pull specific data from API's
- Display API data on all footers
- Stub all tests to prevent API calls during rspec.

## Checklist
- [x] Tests written where required
- [x] All tests passing
- [x] SimpleCov at 100%
    - SimpleCov shows sub 100% coverage due to API service tests being suspended to prevent reaching request limit. Those are the only lines of code not covered.

closes #1
closes #2
closes #3
closes #4

### CODE REVIEW

File: spec/services/github_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#
+#     describe '#closed_pulls' do
+#       xit 'returns all closed pulls from the repo' do
+#         actual = GithubService.closed_pulls
+#         expect(actual[0]).to have_key(:id)
+#         expect(actual[0]).to have_key(:merged_at)
+#       end
+#     end
+#   end
+# end
```
Comment (aliyamerali) [inline spec/services/github_service_spec.rb:29]: All looks good! Any reason you commented out this full test vs just leaving them as skipped to reduce API pings?

**Review by aliyamerali — APPROVED** [review]
Thanks for adding the stubbing across tests! No meaningful comments/changes to add!

File: spec/services/github_service_spec.rb (hunk 1 above)
Comment (LawrenceWhalen) [reply spec/services/github_service_spec.rb:29]: The color of the skipped test is a really hideous goldenrod...

File: app/models/transaction.rb [hunk 2]
```diff
@@ -2,4 +2,5 @@ class Transaction < ApplicationRecord
   enum result: [:failure, :success]
 
   belongs_to :invoice
+
```
Comment (mkrumholz) [inline app/models/transaction.rb:5]: This is probably an eternal war between us where I remove the empty line at the end and then you put it back in. You win this round...

File: app/poros/github_contributors.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+  def self.contributors_info
+    github_contributor_data = GithubService.contributors_info
+    github_contributor_data.each_with_object({}) do |contributor, hash|
+      if [26797256, 78388882, 5446926, 78574189].include?(contributor[:id])
+        hash[contributor[:id]] = {name: contributor[:name], contributions: contributor[:contributions]}
+      end
+    end
+  end
+end
```
Comment (mkrumholz) [inline app/poros/github_contributors.rb:11]: Yay! I think this looks really nice. Proud of us as a team.

File: app/poros/github_pull_requests.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class GithubPullRequests
+
+  def self.merged_pulls
+    github_pull_requests = GithubService.closed_pulls
+    github_pull_requests.count do |pr|
```
Comment (mkrumholz) [inline app/poros/github_pull_requests.rb]: In reality, I'd probably throw out the variable here to streamline. Nit-picky and def a style question.

File: app/poros/github_repo.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class GithubRepo
+
+   def self.name
+     github_data = GithubService.repo_info
+     github_data[:name]
```
Comment (mkrumholz) [inline app/poros/github_repo.rb]: same deal here -- I don't think we need the variable and I find it to be more clutter than clarity

File: app/services/github_service.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    JSON.parse(body, symbolize_names: true)
+  end
+
+  def self.closed_pulls
+    response = Faraday.get "https://api.github.com/repos/LawrenceWhalen/little-esty-shop/pulls?state=closed"
+
+    body = response.body
+    JSON.parse(body, symbolize_names: true)
+  end
+end
```
Comment (mkrumholz) [inline app/services/github_service.rb:23]: I wonder if there's a world in which we just make a method for all of these and then pass in different links. Doesn't work with our current structure, but there's a lot of repeating code here 🧐

File: spec/features/admin/index_spec.rb [hunk 7]
```diff
@@ -2,6 +2,20 @@
...
+      ])
+      allow(GithubService).to receive(:closed_pulls).and_return([
+        {id: 0101010011, name: 'Molly', merged_at: 7},
+        {id: 01011230011, name: 'Sa',merged_at: 80},
+        {id: 01011230011, name: 'Sa', merged_at: nil}
+      ])
+      allow(GithubService).to receive(:repo_info).and_return({
+          name: 'little-esty-shop'
+      })
+    end
```
Comment (mkrumholz) [inline spec/features/admin/index_spec.rb:18]: Thanks for taking the time to do this!

**Review by mkrumholz — APPROVED** [review]
Looked through everything here and as long as the tests are all running without hitting the API, I think we're good to merge. Left some small and nit-picky comments mostly for curiosity's sake.

File: app/poros/github_contributors.rb (hunk 3 above)
Comment (LawrenceWhalen) [reply app/poros/github_contributors.rb:11]: Together we are strong!

File: app/models/transaction.rb (hunk 2 above)
Comment (LawrenceWhalen) [reply app/models/transaction.rb:5]: Pry it from my cold dead hands!

File: app/services/github_service.rb (hunk 6 above)
Comment (LawrenceWhalen) [reply app/services/github_service.rb:23]: Good Idea for a refactor, I might work on that today.

**Pull request merged**